council/llm/ollama_llm.py

```diff
@@ -0,0 +1,144 @@
+from __future__ import annotations
+
+from typing import Any, Final, List, Mapping, Optional, Sequence, Union
+
+from council.contexts import Consumption, LLMContext
+from council.llm import (
+    LLMBase,
+    LLMConfigObject,
+    LLMConsumptionCalculatorBase,
+    LLMCostCard,
+    LLMMessage,
+    LLMProviders,
+    LLMResult,
+    OllamaLLMConfiguration,
+    TokenKind,
+)
+from council.utils.utils import DurationManager
+from ollama import Client
+from ollama._types import Message, Options
+
+
+class OllamaConsumptionCalculator(LLMConsumptionCalculatorBase):
+    DURATION_KEYS: Final[List[str]] = ["prompt_eval_duration", "eval_duration", "load_duration", "total_duration"]
+
+    def get_ollama_consumptions(self, duration: float, response: Mapping[str, Any]) -> List[Consumption]:
+        return (
+            self.get_ollama_base_consumptions(duration)
+            + self.get_ollama_prompt_consumptions(response)
+            + self.get_ollama_duration_consumptions(response)
+        )
+
+    def get_ollama_base_consumptions(self, duration: float) -> List[Consumption]:
+        return [Consumption.call(1, self.model), Consumption.duration(duration, self.model)]
+
+    def get_ollama_prompt_consumptions(self, response: Mapping[str, Any]) -> List[Consumption]:
+        if not all(key in response for key in ["prompt_eval_count", "eval_count"]):
+            return []
+
+        prompt_tokens = response["prompt_eval_count"]
+        completion_tokens = response["eval_count"]
+        return [
+            Consumption.token(prompt_tokens, self.format_kind(TokenKind.prompt)),
+            Consumption.token(completion_tokens, self.format_kind(TokenKind.completion)),
+            Consumption.token(prompt_tokens + completion_tokens, self.format_kind(TokenKind.total)),
+        ]
+
+    def get_ollama_duration_consumptions(self, response: Mapping[str, Any]) -> List[Consumption]:
+        if not all(key in response for key in self.DURATION_KEYS):
+            return []
+
+        # from nanoseconds to seconds
+        return [Consumption.duration(response[key] / 1e9, f"{self.model}:ollama_{key}") for key in self.DURATION_KEYS]
```

Comment:
Let's get rid of `_ollama` as we are in the context of `OllamaConsumptionCalculator`